### DISCUSSION TITLE
How to add arm-none-eabi-gcc compile warnings as report

### DISCUSSION THREAD

**Discussion by LorandMarton** [Ericsson/codechecker#4256] [Q&A]
Hi everyone,
The tool is fantastic, easily generating reports for Clang Static Analyzer (clangsa), clang-tidy, cppcheck, and the GCC static analyzer. These reports are easily viewable on the web server UI, which is incredibly convenient.

I would like to add the arm-none-eabi-gcc compile warnings as a report alongside these.
Has anyone managed to integrate arm-none-eabi-gcc warnings into this reporting system?

Any advice or pointers would be greatly appreciated.

Thank you!

**Comment by LorandMarton** [comment]
My solution for now is as follows:
add a compilation flag: -fdiagnostics-format=sarif-file
running a Python script on all .sarif files: report-converter -t gcc -o ../dir/reports/gcc ./file.c.sarif
then: CodeChecker store on the server

The problem is that I can't define another Analyzer Name like **gcc_compile**, but I can at least filter it because the severity of the warnings is **Unspecified**.

Is there a better way?

**Comment by bruntib** [comment]
Hi @LorandMarton,

Do I understand correctly, that you'd like to provide the path of a specific gcc analyzer binary during CodeChecker analysis? Would you think, this is a solution to the problem?

`CC_ANALYZER_BIN=gcc:/path/to/your/arm-none-eabi-gcc CodeChecker analyze compile_commands.json ...`

**Reply by LorandMarton** [reply]
Thank you for your response.

That part is working well. I have added these configurations to the path:
```
CC_LOGGER_GCC_LIKE gcc:arm-none-eabi-gcc
CC_ANALYZER_BIN gcc:arm-none-eabi-gcc
```
Additionally, arm-none-eabi-gcc is in the path.

When I execute the CodeChecker analyze compile_commands.json ..., I include the -e gcc flag to get the results from the GCC Static Analyzer.

What I aim to achieve is capturing the compiler errors, warnings, and notes produced by arm-none-eabi-gcc and storing them as reports under a different "analyzer" name. From my understanding, this isn't currently possible with the existing setup. In the Makefile, certain warning flags are enabled.

I found a way to capture the compiler warnings by adding `-fdiagnostics-format=sarif-file` to the Makefile to produce SARIF files, which can be converted to reports using report-converter. However, these reports are categorized under the GCC analyzer name, as shown in the screenshot below. The GCC Static Analyzer report and the compile warnings are two different.

![image](https://github.com/Ericsson/codechecker/assets/31390930/2121e677-8f2c-4a7d-815a-19ec3e358d6e)

**Reply by bruntib** [reply]
Now I understand the problem. Unfortunately(?) this is a conceptual decision of CodeChecker what to handle as a checker. Currently, we consider "-Wanalyzer-...." warnings as checkers in case of gcc. Not only CodeChecker captures the diagnostic messages of analyzer tools, but transforms them to a unique name. For example, `-Wanalyzer-double-free` becomes `gcc-double-free`. This diagnostic warning can be enabled with this transformed name. We decided to maintain the list of checkers of all analyzers for certain reasons. It could be a valid request to handle traditional gcc compiler warnings as CodeChecker checkers, but we haven't done it.

Would you think, that it would be necessary? I mean, clang analyzer is also emitting the vast majority of these compiler warnings, though the set of warnings in clang and gcc are most probably disjoint. Can you give an example on a gcc warning which doesn't exist as clang warning? If there is no such warning, then the solution of this problem would be collecting compiler warnings through clang-tidy. You may see, that we keep track of compiler warnings with name `clang-diagnostic-...`:

`CodeChecker checkers | grep clang-diagnostic`
